avocado/utils/download.py

```
@@ -83,24 +86,86 @@ def _url_download(url, filename, data):
     """
-    process = Process(target=_url_download, args=(url, filename, data))
-    log.info("Fetching %s -> %s", url, filename)
-    process.start()
-    process.join(timeout)
-    if process.is_alive():
-        process.terminate()
-        process.join()
-        raise OSError("Aborting downloading. Timeout was reached.")
+    headers = urllib.request.urlopen(
+        urllib.request.Request(url, method="HEAD"), timeout=10
+    ).headers  # Using HEAD method to get the content length
```

Ruff (0.14.6)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)